dcos/mesos.py

```diff
@@ -442,6 +459,35 @@ def tasks(self, fltr=None, completed=False):
 
         return tasks
 
+    # TODO: Currently waiting on this mapping in state.json to be present.
+    def get_container_id(self, task_id):
+        """Returns the container ID for a task ID matching `task_id`
+
+        :param task_id: The task ID which will be mapped to container ID
+        :type task_id: str
+        :returns: the container id associated with task id
+        :rtype: str
+        """
+        if not task_id:
+            #TODO, make this more detailed
+            raise DCOSException("No task id passed to get container_id")
+
+
+        #actually compare the task_id to the task we're looking through
+        #Will need checks to make sure each level exists
+        for framework in self.state()['frameworks']:
+            for task in framework['tasks']:
+                if 'container_id' in task['statuses'][0]:
+                    return task['statuses'][0]['container_id']['value']
```

Comment:
we need to be a little stricter in checks here. Python will explode if any of the earlier dict values  are `None`.